src/Configuration/src/SpringBootBase/SpringBootEnvProvider.cs

```diff
@@ -14,59 +15,42 @@ namespace Steeltoe.Extensions.Configuration.SpringBoot
     /// <summary>
     /// Configuration provider that expands the contents of SPRING_APPLICATION_JSON's Spring-style '.' delimited configuration key/value pairs to .NET compatible form
     /// </summary>
-    public class SpringBootEnvProvider : ConfigurationProvider
+    public class SpringBootEnvProvider : JsonStreamConfigurationProvider
     {
-        internal ILogger<SpringBootEnvProvider> _logger;
-
         private const string SPRING_APPLICATION_JSON = "SPRING_APPLICATION_JSON";
-
-        internal string SpringApplicationJson { get; set; }
+        private readonly string _springApplicationJson;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="SpringBootEnvProvider"/> class.
         /// </summary>
-        /// <param name="logFactory">the logger factory to use</param>
-        public SpringBootEnvProvider(ILoggerFactory logFactory = null)
+        /// <param name="springApplicationJson"> The Json string to parse </param>
+        public SpringBootEnvProvider(string springApplicationJson = null)
+            : base(new JsonStreamConfigurationSource())
         {
-            _logger = logFactory?.CreateLogger<SpringBootEnvProvider>();
-            SpringApplicationJson = Environment.GetEnvironmentVariable(SPRING_APPLICATION_JSON);
+            _springApplicationJson = springApplicationJson;
         }
 
         /// <summary>
-        /// Expands and loads the new keys
+        /// Translates and loads the new keys
         /// </summary>
         public override void Load()
         {
-            try
+            var json = _springApplicationJson ?? Environment.GetEnvironmentVariable(SPRING_APPLICATION_JSON);
+            if (!string.IsNullOrEmpty(json))
             {
-                if (!string.IsNullOrEmpty(SpringApplicationJson))
+                Source.Stream = GetMemoryStream(json);
+                base.Load();
+                var keys = new List<string>(Data.Keys);
+                foreach (var key in keys)
```

Comment:
Wondering if sharing this logic (spring command reformatting) between SpringBootEnvProvider and SpringBootCmdProvider in a common place makes sense. However, looks like `SpringBootCmdProvider` does a Spring prefix check, so this may not make sense if the check can't apply to both.

Reply:
Yes it would probably make sense to reuse but given that difference, it was too trivial. There is another usecase we are not addressing here (checking all env vars for spring. ) when it might make sense to centralize that logic. But for this thinking of going as is for now.

Reply:
Sounds good to me!